tests/types/npm-test.json

```diff
@@ -254,6 +254,316 @@
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the name starts with a dot"
+    },
+    {
+      "description": "Negative Test: An npm purl with a name starting with an underscore should fail",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/_foobar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the name starts with an underscore"
+    },
+    {
+      "description": "Edge Case Test: Version with pre-release and build metadata should be preserved",
+      "test_group": "advanced",
+      "test_type": "roundtrip",
+      "input": "pkg:npm/foobar@1.2.3-alpha.1+build.456",
+      "expected_output": "pkg:npm/foobar@1.2.3-alpha.1+build.456",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Positive Test: Package name with dots and hyphens is valid",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/my-package.js@1.0.0",
+      "expected_output": {
+        "type": "npm",
+        "namespace": null,
+        "name": "my-package.js",
+        "version": "1.0.0",
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Normalization Test: Both scope and name with uppercase letters are lowercased",
+      "test_group": "advanced",
+      "test_type": "roundtrip",
+      "input": "pkg:npm/%40MyScope/MyPackage@1.0.0",
+      "expected_output": "pkg:npm/%40myscope/mypackage@1.0.0",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Negative Test: A purl with only a type should fail parsing",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because name is a required component"
+    },
+    {
+      "description": "Negative Test: An npm purl with a unicode character (emoji) in the name should fail parsing",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/foo%F0%9F%98%83bar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the name contains characters not allowed by npm naming rules"
+    },
+    {
+      "description": "Negative Test: An npm purl with a unicode character (emoji) in the scope should fail parsing",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/%40foo%F0%9F%98%83/bar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the scope contains characters not allowed by npm naming rules"
+    },
```

Comment:
Again, PURL does not define validation rules to forbid this so the tests should not test for that validation.

Reply:
npm does forbid, though. I agree that these shouldn't be in the PURL-spec tests but must be present in SBOM tools at least.